File: PHP/Behavioral/Memento/Example3/Originator.php
<?php

include 'Memento.php';
class Originator
{
    private $state;

    public function setState($state) {
        $this->state = $state;
    }

    public function getState() {
        return $this->state;
    }

    public function saveStateToMemento() {
        return new Memento($this->state);
    }

    public function getStateFromMemento($memento) {
        $this->state = $memento->getState();
    }
}

File: Java/Behavioral/Memento/Example3/Memento/src/Originator.java
public class Originator {
    private String state;

    // Creates memento and stores the current state
    public Memento saveStateToMemento() {
        return new Memento(state);
    }

    // Get the state from the memento and set it to the current state
    public void getStateFromMemento(Memento memento) {
        state = memento.getState();
    }

    public void setState(String state) {
        this.state = state;
    }

    public String getState() {
        return state;
    }
}
